### ISSUE TITLE
Modify S4423: Weak SSL/TLS protocols should not be detected when using AWS Cfn API Gateway.

### ISSUE THREAD

**Issue by sebastien-andrivet-sonarsource** [SonarSource/SonarJS#4027]
# What:

This rule detects the usage of weak SSL or TLS protocols especially when no explicit version of TLS is specified. Currently, it raises a false positive for AWS Cfn API Gateway V1. This API is using TLS 1.2 by default. Non-Cfn Gateway API and API Gateway V2 are currently correctly detected.

# Detection:

* Fully qualified class names:

- `aws-cdk-lib.aws_apigatewayv2.CfnDomainName`
- `aws-cdk-lib.aws_apigateway.CfnDomainName`
- `aws-cdk-lib.aws_apigateway.DomainName`

* Properties to check

- For `aws_apigatewayv2.CfnDomainName`: `domainNameConfigurations.securityPolicy`
- For `aws_apigateway.CfnDomainName`: `securityPolicy`
- For `aws_apigateway.DomainName`: `securityPolicy`

* Raise an issue if:

- For `aws_apigateway.CfnDomainName`: raise an issue if `securityPolicy` is set to `'TLS_1_0'`
- For `aws_apigateway.DomainName`: raise an issue if `securityPolicy` is set to `'TLS_1_0'`

* Do not raise an issue if:

- For `aws_apigatewayv2.CfnDomainName`: do not raise an issue if `securityPolicy` is set to `'TLS_1_2'` or higher or if `securityPolicy` is missing.
- For `aws_apigateway.CfnDomainName`: do not raise an issue if `securityPolicy` is set to `'TLS_1_2'` or higher or if `securityPolicy` is missing.
- For `aws_apigatewayv2.CfnDomainName`: do not raise an issue if `securityPolicy` is set to `'TLS_1_2'` or higher or if `securityPolicy` is missing.

# Examples:

- Noncompliant:

```typescript
    const cdn1 = new agw.CfnDomainName(this, 'Explicitly1_0NonCompliantDomainName', {
        regionalCertificateArn: certificateArn,
        domainName: ''arn:aws:acm:eu-central-1:501215020883:certificate/f3511e63-19de-4daa-b34a-775896704789'',
        endpointConfiguration: {
          types: ['REGIONAL']
        },
        securityPolicy: 'TLS_1_0', // NonCompliant
    });
```

- Compliant:

```typescript
    const cdn2 = new agw.CfnDomainName(this, 'Explicitly1_2CompliantDomainName', {
        regionalCertificateArn: certificateArn,
        domainName: 'compliant-cfn.example.com',
        endpointConfiguration: {
          types: ['REGIONAL']
        },
        securityPolicy: 'TLS_1_2', // Compliant
    });

    const cdn3 = new agw.CfnDomainName(this, 'ImplicitlyCompliantDomainName', {
        regionalCertificateArn: certificateArn,
        domainName: 'default-cfn.example.com',
        endpointConfiguration: {
          types: ['REGIONAL']
        }
    }); // Compliant: defaults to SecurityPolicy.TLS_1_2

```

# RSPEC

The relevant RSPEC is: https://github.com/SonarSource/rspec/pull/2520

**Comment by Wohops** [comment]
Migrated to Jira. New ticket number: JS-83
